### ISSUE TITLE
Benchmark coverage limited to LintContent; CPU-bound regressions invisible without profiling

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#393]
## Overview

The current benchmarks (`BenchmarkFullLinting`, `BenchmarkFullLinting_ExtraLarge`) only exercise `LintContent`. The CI comparison tracks `ns/op`, `B/op`, and `allocs/op` via `benchmem`. Two blind spots follow from this.

## Problem 1: Benchmark scope

`LintContent` covers:

```
StripFrontmatter → strings.Split → preprocess.Scan → collectLineErrors
```

The following paths are entirely invisible to the benchmark:

| Path | Why it matters |
|---|---|
| `file.ExpandPaths` | `filepath.WalkDir` + syscalls; I/O bound, not allocation bound |
| `file.ReadFile` | `os.ReadFile`; same |
| `lint.Run` goroutine fan-out | `sync.Mutex` contention under concurrent file load |
| `output.Formatter` (text/JSON) | String building, JSON serialization |

A regression in any of these paths is undetectable by the current benchmark.

## Problem 2: allocs/op and B/op do not catch CPU-bound regressions

`allocs/op` and `B/op` signal memory pressure. When a rule adds a CPU-intensive operation per line — a regex match, repeated `strings.TrimSpace`, a complex string scan — `ns/op` rises while `allocs/op` and `B/op` stay flat. The current CI threshold check (`⚠️ ≥+10%`, `❌ ≥+50%`) on those two metrics would not flag the regression; only `ns/op` would show it, and the diagnostic tool to understand it is a CPU profile, not allocation counts.

## Tasks

- [x] #394
- [x] #395
